application/models/Template.php

```diff
@@ -147,8 +147,8 @@ public static function templateNameFilter($sTemplateName)
         /* Validate if template is OK in user dir, DIRECTORY_SEPARATOR not needed "/" is OK */
         $oTemplate = self::model()->findByPk($sTemplateName);
 
-        if (is_object($oTemplate) && $oTemplate->checkTemplate() && (self::checkTemplateXML($oTemplate->folder))) {
-            self::$aNamesFiltered[$sTemplateName] = $sTemplateName;
+        if (is_object($oTemplate) && $oTemplate->checkTemplate() && self::checkTemplateXML($oTemplate->folder)) {
+            self::$aNamesFiltered[$sTemplateName] = $oTemplate->folder;
```

Comment:
I am not to sure about this.
I would like to have someone else review this as well.

By returning the foldername, somehow:
1 - the intention of the method seems to be changed.
2 - folder and templatename may match so far, but i have my doubts about exchanging them so freely.

Are we sure the source of the problem is here?

Reply:
> Are we sure the source of the problem is here?

Yes : on linux MyTheme != mytheme, but on DB MyTheme == mytheme

The other way are are to check MyTheme != mytheme on DB too.

Reply:
I add `$oTemplate->name === $sTemplateName` in comparaiosn.

For information name and folder are same in object (we can remove folder and use name)